### PR TITLE
Iryna k task5 implement authentication and authorization

### PR DESCRIPTION
**Pull request by rinchan603**
#5 
Instruction for initializing connect to DB:
1. **Create empty DB "Elevel"** in MSSQL Server
2. **Update** connection string in **appsettings.json**
3. **Open Package Manager Console** in VS and select **Default project: Elevel.Infrastructure.Persistence**
4. **Run command** from the console: **update-database**

### CODE REVIEW

File: Elevel/Elevel.Api/Startup.cs [hunk 1]
```diff
@@ -28,14 +38,63 @@ public Startup(IConfiguration configuration)
         // This method gets called by the runtime. Use this method to add services to the container.
         public void ConfigureServices(IServiceCollection services)
         {
+            //Configuration from AppSettings
+            services.Configure<TokenConfiguration>(Configuration.GetSection("JWT"));
+            //User Manager Service
+            services.AddIdentity<ApplicationUser, IdentityRole>().AddEntityFrameworkStores<ApplicationDbContext>();
```
Comment (parys) [inline Elevel/Elevel.Api/Startup.cs:44]: here should be IdentityRole<Guid> I think

File: Elevel/Elevel.Infrastructure.Services/Implementation/UserService.cs [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+            _jwt = jwt.Value;
+        }
+
+        public async Task<AuthenticationModel> GetTokenAsync(TokenRequestModel model)
+        {
+            var authenticationModel = new AuthenticationModel();
+            var user = await _userManager.FindByEmailAsync(model.Email);
+            if (user == null)
+            {
+                authenticationModel.IsAuthenticated = false;
```
Comment (parys) [inline Elevel/Elevel.Infrastructure.Services/Implementation/UserService.cs:35]: you can set this value as false at initialization (row 31) but I think it false by default

File: Elevel/Persistence/Context/ApplicationDbContextSeed.cs [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+
+namespace Elevel.Infrastructure.Persistence.Context
+{
+    public class ApplicationDbContextSeed
+    {
+        public static async Task SeedEssentialsAsync(UserManager<ApplicationUser> userManager,
+            RoleManager<IdentityRole> roleManager)
+        {
+            //Seed Roles
+            await roleManager.CreateAsync(new IdentityRole(Authorization.Roles.Administrator.ToString()));
```
Comment (parys) [inline Elevel/Persistence/Context/ApplicationDbContextSeed.cs:14]: before you should check, that DB is not contain anything, in other case each run you will add copy

Comment (rinchan603) [reply Elevel/Persistence/Context/ApplicationDbContextSeed.cs]: I checked - no duplicates are added. I think that UserManager and RoleManager already have built-in check for adding duplicates.

**Review by parys — APPROVED** [review]

File: Elevel/Elevel.Api/Startup.cs (hunk 1 above)
Comment (rinchan603) [reply Elevel/Elevel.Api/Startup.cs]: Done.

File: Elevel/Elevel.Infrastructure.Services/Implementation/UserService.cs (hunk 2 above)
Comment (rinchan603) [reply Elevel/Elevel.Infrastructure.Services/Implementation/UserService.cs]: Done.

**Review by parys — APPROVED** [review]

File: Elevel/Domain/Authorization.cs [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+{
+    public class Authorization
+    {
+        public enum Roles
+        {
+            Administrator,
+            HumanResourceManager,
+            Coach,
+            User
+        }
```
Comment (DShcharbakou) [inline Elevel/Domain/Authorization.cs:15]: I would move this enum into a separate file. I think it will be used not only in this file.

Comment (rinchan603) [reply Elevel/Domain/Authorization.cs:15]: Done.

**Review by DShcharbakou — APPROVED** [review]

File: Elevel/Domain/Enums/UserRole.cs [hunk 5]
```diff
@@ -0,0 +1,10 @@
+﻿namespace Elevel.Domain.Enums
+{
+    public enum UserRole
+    {
+        Administrator,
+        HumanResourceManager,
```
Comment (andreyrogov8) [inline Elevel/Domain/Enums/UserRole.cs:6]: why not just "HR"?

File: Elevel/Domain/Models/Test.cs [hunk 6]
```diff
@@ -10,39 +10,51 @@ namespace Elevel.Domain.Models
...
 
         public Guid? SpeakingId { get; set; }
+
         //public Speaking Speaking { get; set; }
+
         public string SpeakingAnswerReference { get; set; }
-        public int? SpeakingMark { get; set; }
 
+        public int? SpeakingMark { get; set; }
+
```
Comment (andreyrogov8) [inline Elevel/Domain/Models/Test.cs:57]: Do you really need these spaces?
I made it that way for grouping it

File: Elevel/Elevel.Api/appsettings.json [hunk 7]
```diff
@@ -6,8 +6,17 @@
...
     "DefaultConnection": "Server=PC\\SQLEXPRESS;Database=Elevel;Trusted_Connection=True;MultipleActiveResultSets=true"
+  },
+
+  "JWT": {
+    "key": "C1CF4B7DC4C4175B6618DE4F55CA4",
+    "Issuer": "SecureApi",
+    "Audience": "SecureApiUser",
+    "DurationInMinutes": 65
   }
 }
```
Comment (andreyrogov8) [inline Elevel/Elevel.Api/appsettings.json:22]: Why don't you use SQL client?

Comment (rinchan603) [reply Elevel/Elevel.Api/appsettings.json:22]: This parameter doesn't need here. EF will do that.

File: Elevel/Domain/Models/Test.cs (hunk 6 above)
Comment (rinchan603) [reply Elevel/Domain/Models/Test.cs:57]: This spaces improve reading of code by my experience.

File: Elevel/Domain/Enums/UserRole.cs (hunk 5 above)
Comment (rinchan603) [reply Elevel/Domain/Enums/UserRole.cs:6]: I decided that using of meaningfull values is better than abriviature.

**Review by parys — APPROVED** [review]

**Pull request merged**